core/src/main/java/io/questdb/std/json/SimdJsonParser.java

```diff
@@ -0,0 +1,302 @@
+        return queryPointerInt(
+                impl,
+                json.ptr(),
+                json.size(),
+                json.tailPadding(),
+                pointer.ptr(),
+                pointer.size(),
+                result.ptr()
+        );
+    }
+
+    public long queryPointerLong(
+            DirectUtf8Sequence json,
+            DirectUtf8Sequence pointer,
+            SimdJsonResult result
+    ) {
+        assert json.tailPadding() >= SIMDJSON_PADDING;
+        return queryPointerLong(
+                impl,
+                json.ptr(),
+                json.size(),
+                json.tailPadding(),
+                pointer.ptr(),
+                pointer.size(),
+                result.ptr()
+        );
+    }
+
+    public short queryPointerShort(
+            DirectUtf8Sequence json,
+            DirectUtf8Sequence pointer,
+            SimdJsonResult result
+    ) {
+        assert json.tailPadding() >= SIMDJSON_PADDING;
+        return queryPointerShort(
+                impl,
+                json.ptr(),
+                json.size(),
+                json.tailPadding(),
+                pointer.ptr(),
+                pointer.size(),
+                result.ptr()
+        );
+    }
+
+    public void queryPointerUtf8(
+            DirectUtf8Sequence json,
+            DirectUtf8Sequence pointer,
+            SimdJsonResult result,
+            DirectUtf8Sink dest,
+            int maxSize
+    ) {
+        assert json.tailPadding() >= SIMDJSON_PADDING;
+        final long nativeByteSinkPtr = dest.borrowDirectByteSink().ptr();
+        assert dest.capacity() - dest.size() >= maxSize;  // Without this guarantee we'd need to close `NativeByteSink.close`.
+        queryPointerUtf8(
+                impl,
+                json.ptr(),
+                json.size(),
+                json.tailPadding(),
+                json.isAscii(),
+                pointer.ptr(),
+                pointer.size(),
+                result.ptr(),
+                nativeByteSinkPtr,
+                maxSize
+        );
+    }
+
+    /**
+     * Generic call that can access any JSON value.
+     * Consult the `result`'s `getError()`, `getType()` and `getNumberType()` for processing the data.
+     * <p>
+     * For example, if the path points to a JSON number extracted as a double:
+     * * `result.getError()` will be `SimdJsonError.SUCCESS`
+     * * `result.getType()` will be `SimdJsonType.NUMBER`
+     * * `result.getNumberType()` will be `SimdJsonNumberType.FLOATING_POINT_NUMBER`
+     * * `queryPointerValue`'s returned long needs to be converted to a double by calling `Double.longBitsToDouble`.
+     * <p>
+     * On a `null`, the `result.getType()` will be `SimdJsonType.NULL` and returned value 0.
+     * <p>
+     * On a boolean, the `result.getType()` will be `SimdJsonType.BOOLEAN` and returned value reliably 0 or 1.
+     * <p>
+     * Arrays and objects are returned as raw JSON strings.
```

Comment:
Add a note that when returning a JSON string, the `ascii` flag will not be set even when encoutering multibyte characters.